crates/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -689,8 +689,19 @@ impl<'interner> Monomorphizer<'interner> {
             HirType::Function(args, ret, env) => {
                 let args = vecmap(args, Self::convert_type);
                 let ret = Box::new(Self::convert_type(ret));
-                let env = Box::new(Self::convert_type(env));
-                ast::Type::Function(args, ret, env)
+                let env = Self::convert_type(env);
+                match &env {
+                    ast::Type::Unit => ast::Type::Function(args, ret, Box::new(env)),
+                    ast::Type::Tuple(_elements) => ast::Type::Tuple(vec![
+                        env.clone(),
+                        ast::Type::Function(args, ret, Box::new(env)),
+                    ]),
```

Comment:
Do we still need the `env` field on the monomorphized function type if we're translating them directly to a tuple around a function here anyway? Might want to look into removing it in a future PR if it is no longer needed.

Reply:
correct! I did replace it with a `is_closure` bool field succesfuly locally.

Should i add that as a commit to this PR, or should i open a new one after merging this?

Reply:
Why would we even need an `is_closure` field if the function is already closure-converted?